File: MicroBlog/MicroBlog/Models/Message.cs
using System.Xml.Serialization;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace MicroBlog.Models;

[Serializable]
public class Message
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string? Id { get; set; }

    [XmlAttribute("Id")]
    public int XmlId { get; set; }

    [XmlAttribute("PostTypeId")]
    public int PostTypeId { get; set; }
    
    [XmlAttribute("AcceptedAnswerId")]
    public int AcceptedAnswerId { get; set; }
    
    [XmlAttribute("CreationDate")]
    public string CreationDate { get; set; } = null!;
    
    [XmlAttribute("Score")]
    public int Score { get; set; }
    
    [XmlAttribute("ViewCount")]
    public int ViewCount { get; set; }
    
    [XmlAttribute("Body")]
    public string Body { get; set; } = null!;
    
    [XmlAttribute("OwnerUserId")]
    public int OwnerUserId { get; set; }
    
    [XmlAttribute("LastEditorUserId")] 
    public int LastEditorUserId { get; set; }
    
    [XmlAttribute("LastEditDate")]
    public string LastEditDate { get; set; } = null!;
    
    [XmlAttribute("LastActivityDate")]
    public string LastActivityDate { get; set; } = null!;
    
    [XmlAttribute("Title")]
    public string Title { get; set; } = null!;
    
    [XmlAttribute("Tags")]
    public string Tags { get; set; } = null!;
    
    [XmlAttribute("AnswerCount")]
    public int AnswerCount { get; set; }
    
    [XmlAttribute("CommentCount")]
    public int CommentCount { get; set; }
    
    [XmlAttribute("ContentLicense")]
    public string ContentLicense { get; set; } = null!;
}

[Serializable]
[XmlRoot("MessageCollection")]
public class MessageCollection
{
    [XmlArray("posts")]
    [XmlArrayItem("row", typeof(Message))]
    public Message[]? Messages { get; set; }
}

File: MicroBlog/DataLoaderGo/models/message.go
package models

import (
	"encoding/xml"
)

type Message struct {
	Id               string   `bson:"_id,omitempty" json:"id"`
	XmlId            int      `bson:"xml_id" json:"xmlId" xml:"Id,attr"`
	PostTypeId       int      `bson:"post_type_id" json:"postTypeId" xml:"PostTypeId,attr"`
	AcceptedAnswerId int      `bson:"accepted_answer_id" json:"acceptedAnswerId" xml:"AcceptedAnswerId,attr"`
	CreationDate     timeType `bson:"creation_date" json:"creationDate" xml:"CreationDate,attr"`
	Score            int      `bson:"score" json:"score" xml:"Score,attr"`
	ViewCount        int      `bson:"view_count" json:"viewCount" xml:"ViewCount,attr"`
	Body             string   `bson:"body" json:"body" xml:"Body,attr"`
	OwnerUserId      int      `bson:"owner_user_id" json:"ownerUserId" xml:"OwnerUserId,attr"`
	LastEditorUserId int      `bson:"last_editor_user_id" json:"lastEditorUserId" xml:"LastEditorUserId,attr"`
	LastEditDate     timeType `bson:"last_edit_date" json:"lastEditDate" xml:"LastEditDate,attr"`
	LastActivityDate timeType `bson:"last_activity_date" json:"lastActivityDate" xml:"LastActivityDate,attr"`
	Title            string   `bson:"title" json:"title" xml:"Title,attr"`
	Tags             string   `bson:"tags" json:"tags" xml:"Tags,attr"`
	AnswerCount      int      `bson:"answer_count" json:"answerCount" xml:"AnswerCount,attr"`
	CommentCount     int      `bson:"comment_count" json:"commentCount" xml:"CommentCount,attr"`
	ContentLicense   string   `bson:"content_license" json:"contentLicense" xml:"ContentLicense,attr"`
}

type MessageCollection struct {
	XmlName xml.Name  `xml:"MessageCollection"`
	Posts   PostArray `xml:"posts"`
}

type PostArray struct {
	Messages []Message `xml:"row"`
}
